evaluate/task/task.go

```diff
@@ -32,3 +38,105 @@ var (
 	// IdentifierCodeRepair holds the identifier for the "code repair" task.
 	IdentifierCodeRepair = registerIdentifier("code-repair")
 )
+
+// TaskForIdentifier returns a task based on the task identifier.
+func TaskForIdentifier(taskIdentifier evaltask.Identifier, logger *log.Logger, resultPath string, model model.Model, language language.Language) (task evaltask.Task, err error) {
+	switch taskIdentifier {
+	case IdentifierWriteTests:
+		return NewTaskWriteTests(logger, resultPath, model, language), nil
+	default:
+		return nil, evaltask.ErrTaskUnsupported
+	}
+}
+
+// TaskWriteTests holds the write test task.
+type TaskWriteTests struct {
+	// ResultPath holds the directory path where results should be written to.
+	ResultPath string
+
+	// Language holds the language for which the task should be evaluated.
+	Language language.Language
+	// Model holds the model which the task should be evaluated.
+	Model model.Model
+
+	// Logger holds the logger for this tasks.
+	Logger *log.Logger
+}
+
+var _ evaltask.Task = (*TaskWriteTests)(nil)
+
+// NewTaskWriteTests returns a write test task.
+func NewTaskWriteTests(logger *log.Logger, resultPath string, model model.Model, language language.Language) (task evaltask.Task) {
+	return &TaskWriteTests{
+		ResultPath: resultPath,
+		Language:   language,
+		Model:      model,
+		Logger:     logger,
+	}
+}
```

Comment:
nit: could make extra files for specific tasks... once we have like... 5 of them, this will be one giant spaghetti file.